lib/dynamoid/criteria/key_fields_detector.rb

```diff
@@ -133,6 +136,31 @@ def match_global_secondary_index
           }
         end
       end
+
+      def match_forced_index
+        idx = possilbe_local_secondary_index || possible_global_secondary_index
+
+        # TODO: Handle named index not found
+        # TODO: Check equality condition exists on requested index hash key
```

Comment:
1. I think we should raise exception when index name isn't found, e.g. `Dynamoid::Errors::InvalidIndex`.
2. It's OK to check name here but I would say it makes sense to check it in `Chain#with_index` method.
3. Using forced index name without equality condition may seem useless (because it means Scan operation instead of Query) but it's still a valid case and it could be reasonable for instance when GSI is used and not all attributes are projected - this way the operation will be cheaper. So I don't see any reason to check equality condition at all here.